### PR TITLE
ci: push docker images to both docker hub and GCR

### PR DESCRIPTION
**Pull request by ggilmore**
This PR changes our CI pipeline to push images to both hub.docker.io and [Google Container Registry (GCR)](https://cloud.google.com/container-registry/). GCR has a security scanning feature that we can use to perform checks on all of our published Docker images (we'll still tell our users to download the images from hub.docker.io)

cc @sourcegraph/distribution

### CODE REVIEW

**Comment by ggilmore** [comment]
Hold one one sec, just noticed an issue

File: enterprise/dev/ci/ci/pipeline-steps.go [hunk 1]
```diff
@@ -272,30 +272,37 @@ func addDockerImage(c Config, app string, insiders bool) func(*bk.Pipeline) {
...
-		if app == "server" && c.releaseBranch {
-			cmds = append(cmds,
-				bk.Cmd(fmt.Sprintf("docker tag %s:%s %s:%s-insiders", image, c.version, image, c.branch)),
-				bk.Cmd(fmt.Sprintf("docker push %s:%s-insiders", image, c.branch)),
-			)
-		}
+		for _, image := range []string{dockerHubImage, gcrImage} {
+			if app != "server" || c.taggedRelease || c.patch || c.patchNoTest {
+				cmds = append(cmds,
+					bk.Cmd(fmt.Sprintf("docker tag %s:%s %s:%s", baseImage, c.version, image, c.version)),
```
Comment (uwedeportivo) [inline enterprise/dev/ci/ci/pipeline-steps.go]: i think you're doing the same tagging twice: once for docker hub, once for gcr

i guess tagging is idempotent, but maybe extract the tagging outside the loop ?

Comment (ggilmore) [reply enterprise/dev/ci/ci/pipeline-steps.go]: @uwedeportivo I'm not quite sure that I follow you? We need to create two separate tags to upload them to different registries: one with the `index.docker.io` prefix and another one with the `us.gcr.`io` prefix. `docker push us.gcr.io/sourcegraph-dev/[APP]:[VERSION]` will fail if you haven't tagged an image with that exact name yet.

Comment (uwedeportivo) [reply enterprise/dev/ci/ci/pipeline-steps.go]: oh, i see now. the tag depends on image too, not kust baseImage. my bad. ignore my comment 😄

**Pull request merged**